src/core/include/iDynTree/Core/Utils.h

```diff
@@ -17,6 +17,33 @@
  */
 #define IDYNTREE_UNUSED(var) ((void)var)
 
+/**
+ * \brief Macro to deprecate functions and methods
+ *
+ * see https://blog.samat.io/2017/02/27/Deprecating-functions-and-methods-in-Cplusplus/
+ */
+// C++14
+#if __cplusplus >= 201402L
+  #if defined(__has_cpp_attribute)
+    #if __has_cpp_attribute(deprecated)
+      #define IDYNTREE_DEPRECATED [[deprecated]]
+      #define IDYNTREE_DEPRECATED_WITH_MSG(func, msg) [[deprecated(msg)]] func
```

Comment:
Just to clarify: I did not want to get rid of the message warning function, I suggested to change the existing macro:
~~~
#define IDYNTREE_DEPRECATED_WITH_MSG(func, msg) [[deprecated(msg)]] func 
~~~
to the one (more compatible with the C++14 attribute styles): 
~~~
#define IDYNTREE_DEPRECATED_WITH_MSG(msg) [[deprecated(msg)]] 
~~~
It seems to me that the second version would be easier to use and read. 

However, it is fine for me to merge the PR in the current form.